Fix major merge conflicts in thanks file

xexyl opened the pull request:
Unfortunately the renaming of the thanks-for-fixes.md to thanks-for-help.md along with other changes with the it caused massive merge conflicts. The file was out of order, there were a lot of >>> lines etc.

I had to checkout the old file and update it for the changes I made this morning. Unfortunately this does mean that the <a ..> anchors are no longer there, relying on markdown to do that instead. That will have to be fixed but I also noticed an error in that too. I am not sure if that's because of the merge conflict or not though.

I will have to check that I got everything in order but I think I did as I also added what I had added to it today.

xexyl commented:
Sorry for having to undo the anchors but it was the only way :(

I have to check that everything is there and then I'll be updating it again with more entries being worked on today.

lcn2 commented:
This shows that your tree is not in sync with the repo.

xexyl commented:
> This shows that your tree is not in sync with the repo.

I just pulled from ioccc-src copy and it has the same problem so I don't know that that's the case.

xexyl commented:
Will try something.

xexyl commented:
No it's definitely here. Look at https://github.com/ioccc-src/temp-test-ioccc/blob/master/thanks-for-help.md#2015.

lcn2 commented:
Here is what we will need to do:

Rather than reject this pull request which will impact future pull requests: we are going to quickly accept the set of pull requests and then immdately revert this particular pull request by replacing it with that is in the master now (pre this pull request) with the contents as of commit b85e20aac3c81b96a2d4e2728adf33f190b25364.  Whatever happened to the `thanks-for-help.md` in this particular pull request will be lost.

Stay tuned .. this is happening now ...

xexyl commented:
> Here is what we will need to do:
> 
> Rather than reject this pull request which will impact future pull requests: we are going to quickly accept the set of pull requests and then immdately revert this particular pull request by replacing it with that is in the master now (pre this pull request) with the contents as of commit [b85e20a](https://github.com/ioccc-src/temp-test-ioccc/commit/b85e20aac3c81b96a2d4e2728adf33f190b25364). Whatever happened to the `thanks-for-help.md` in this particular pull request will be lost.
> 
> Stay tuned .. this is happening now ...

But if you look at what I just posted you'll see there IS a problem here.

xexyl commented:
```
>>>>>>> xexyl-duble15-try 
<<<<<<< HEAD

||||||| 6a936532

>>>>>>> xexyl-duble15-try

```

and many more like these. So if you undo this you'll be putting the problem back in place and undoing the additional things I added.

xexyl commented:
It does seem that it might be easier to fix but since you're doing something with it I don't want to try now. It means I'll have to re-add the things and I have no idea what I added. I'd rather fix it then.

lcn2 commented:
With commit 8af017fefba867f6f4ba7c9a1ec0e0246c6c9444 we are back to the state were the "<<<" issue can now be corrected.

Stay tuned ...

xexyl commented:
> > With commit [8af017f](https://github.com/ioccc-src/temp-test-ioccc/commit/8af017fefba867f6f4ba7c9a1ec0e0246c6c9444) we are back to the state were the "<<<" issue can now be corrected.
> > Stay tuned ...
> 
> .. and undid the things I added :(

And I don't know what I added. Can you add them back? Not doing that will slow me down in other things. Well anyway I'm done for the day and this weekend will be slower most likely. I'll at least make a copy of the file so I can have the same text. But as you can see from the above the problem was here, not because my fork was out of sync.

lcn2 commented:
Corrected with commit 8af017fefba867f6f4ba7c9a1ec0e0246c6c9444

lcn2 commented:
See commit e74a5d1c96af8ad4bdbec3c12e8c362d65108838

lcn2 commented:
> > Corrected with commit [8af017f](https://github.com/ioccc-src/temp-test-ioccc/commit/8af017fefba867f6f4ba7c9a1ec0e0246c6c9444)
> 
> 
> 
> But does that include the things I added a bit ago, after this pull request? Well I have a copy of the file anyway so hopefully it won't be too hard to fix. But that's for another time - not until tomorrow. Off to get some food.

With commit e74a5d1c96af8ad4bdbec3c12e8c362d65108838 it should include them.

xexyl commented:
> > > Corrected with commit [8af017f](https://github.com/ioccc-src/temp-test-ioccc/commit/8af017fefba867f6f4ba7c9a1ec0e0246c6c9444)
> > 
> > 
> > But does that include the things I added a bit ago, after this pull request? Well I have a copy of the file anyway so hopefully it won't be too hard to fix. But that's for another time - not until tomorrow. Off to get some food.
> 
> With commit [e74a5d1](https://github.com/ioccc-src/temp-test-ioccc/commit/e74a5d1c96af8ad4bdbec3c12e8c362d65108838) it should include them.

Thanks! Well hope you have a great afternoon. I'll do more tomorrow but as you'll see in another issue it will be a weekend where I'll have less time.

lcn2 commented:
Doing more checking ...

lcn2 commented:
Looks like `thanks-for-help.md` is back to a reasonable state and the meaningful changes have been applied.

Sorry (tm Canada 🇨🇦) for the merge confusion.

xexyl commented:
> Looks like `thanks-for-help.md` is back to a reasonable state and the meaningful changes have been applied.

Thank you.
> 
> Sorry (tm Canada 🇨🇦) for the merge confusion.

Well it's not your fault. I was a bit frustrated but at the situation. I've had numerous problems with git merges as well and it's always annoying. I thought there were many more instances of the problems. If I had realised there weren't I might have been able to fix it first and prevented the problem but unfortunately I did not.

I'm off for food. There will be a time where I might be back at laptop but probably won't do anything here.

So I hope you have a great day!

xexyl commented:
Thank you very much for your effort: I really, really appreciate it!

https://github.com/ioccc-src/temp-test-ioccc/pull/1670/commits/0f3a4b8a68c75b676277435ca0653bd33d7d48f7 made some final fixes and it's now in good shape.

Hope you're sleeping well. I am going to rest .. unless the cat gets on my bed which might very well happen as he's in my room (so is the other one, his mum, but she won't get between my legs if she gets up on the bed which is entirely possible too).
